### PR TITLE
Add Payments claims list

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

We need to display a list of claims that need to be processed within the Payments flow.

## Changes proposed in this pull request

- Add `href` attribute to `Claims::CardComponent`.
- Add the claims list and filters to the payments index page, if any claims fall within the `payment_information_requested` or `payment_information_sent` statuses.
- Add the payments claims show page. `/support/claims/payments/claims/:id`.

## Guidance to review

- The payments claims show page is currently identical to the claims show page. In an upcoming PR, the payments claims show page will have additional information and action buttons.

## Screenshots

CleanShot 2024-12-06 at 13 38 16@2x

### CODE REVIEW

File: app/policies/claims/support/payments/claim_policy.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Claims::Support::Payments::ClaimPolicy < Claims::ApplicationPolicy
+end
```
Comment (Nitemaeric) [inline app/policies/claims/support/payments/claim_policy.rb:2]: Payments-specific actions will be added in an upcoming PR.

File: app/views/claims/schools/claims/show.html.erb [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
       <% if policy(@claim).submit? %>
         <%= govuk_button_to t(".submit"), check_claims_school_claim_path(@school, @claim), method: :get %>
-      <% elsif @claim.submitted? %>
+      <% elsif @claim.submitted_by_id? %>
```
Comment (Nitemaeric) [inline app/views/claims/schools/claims/show.html.erb:21]: This was previously checking the status. All statuses past the `submitted` status should have the `submitted_by_id` and `submitted_at` values.

File: app/views/claims/support/claims/samplings/index.html.erb [hunk 3]
```diff
@@ -12,7 +12,7 @@
     <%= render Claims::Claim::FilterFormComponent.new(filter_form:, statuses: %w[sampling_in_progress sampling_provider_not_approved]) do %>
       <div class="govuk-!-margin-bottom-2">
         <% @claims.each do |claim| %>
-          <%= render Claim::CardComponent.new(claim:) %>
+          <%= render Claim::CardComponent.new(claim:, href: claims_support_claim_path(claim)) %>
```
Comment (Nitemaeric) [inline app/views/claims/support/claims/samplings/index.html.erb:15]: This link is incorrect, but it is necessary for now.

File: spec/system/claims/support/schools/claims/edit_draft_claim_spec.rb [hunk 4]
```diff
@@ -15,9 +15,7 @@
       :claim,
       :draft,
       school:,
-      reference: "12345678",
       provider: best_practice_network_provider,
-      submitted_by: colin,
```
Comment (Nitemaeric) [inline spec/system/claims/support/schools/claims/edit_draft_claim_spec.rb:20]: A claim should never be submitted by a support user. This was incorrect.

File: spec/factories/claims.rb [hunk 5]
```diff
@@ -51,9 +51,26 @@
...
+    end
+
+    trait :payment_information_requested do
+      payment_in_progress
+      status { :payment_information_requested }
+    end
+
+    trait :payment_information_sent do
+      payment_information_requested
+      status { :payment_information_sent }
```
Comment (Nitemaeric) [inline spec/factories/claims.rb:73]: These changes chain claims as if they have been through the previous statuses.

File: app/views/claims/support/claims/payments/claims/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+<% end %>
+
+<div class="govuk-width-container">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <p class="govuk-caption-l"><%= t(".page_caption", reference: @claim.reference) %></p>
+      <h1 class="govuk-heading-l"><%= @claim.school.name %> <%= render Claim::StatusTagComponent.new(claim: @claim) %></h1>
+
+      <% if @claim.submitted_by_id? %>
+        <p class="govuk-body"><%= t(".submitted_by", name: @claim.submitted_by.full_name, date: l(@claim.submitted_on, format: :long)) %></p>
```
Comment (Kizr) [inline app/views/claims/support/claims/payments/claims/show.html.erb:15]:
```suggestion
        <p class="govuk-body"><%= t(".submitted_by", name: @claim.submitted_by_full_name, date: l(@claim.submitted_on, format: :long)) %></p>
```

Could we make use of the existing delegation on the claim model here?

File: config/locales/en/claims/support/claims/payments.yml [hunk 7]
```diff
@@ -5,4 +5,11 @@ en:
         payments:
           index:
             heading: Claims
-            sub_heading: Payments
+            sub_heading:
+              zero: Payments
+              one: Payments (%{count})
```
Comment (Kizr) [inline config/locales/en/claims/support/claims/payments.yml:10]: It'd be nice if we could omit `one` here, is that a possibility or does it fall over?

**Review by Kizr — APPROVED** [review]
Looks solid to me, nothing blocking but a few things to consider 😄

File: config/locales/en/claims/support/claims/payments.yml (hunk 7 above)
Comment (Nitemaeric) [reply config/locales/en/claims/support/claims/payments.yml:10]: Yeah, it falls over 😢

File: app/controllers/claims/support/claims/payments_controller.rb [hunk 8]
```diff
@@ -1,3 +1,45 @@
...
+
+  def index
+    @pagy, @claims = pagy(filtered_claims)
+  end
+
+  private
+
+  def filtered_claims
+    @filtered_claims ||= Claims::ClaimsQuery.call(params: filter_form.query_params)
+                                            .where(status: %i[payment_information_requested payment_information_sent])
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply app/controllers/claims/support/claims/payments_controller.rb:14]: I do. It's reused at least once, so a constant is a good way to keep them aligned.

File: app/components/claim/card_component.html.erb [hunk 9]
```diff
@@ -1,9 +1,9 @@
 <div class="claim-card">
   <div class="claim-card__header">
     <div class="claim-card__header__name govuk-heading-s">
-      <%= govuk_link_to t(".claim_title", reference: claim.reference, school_name: claim.school.name), claims_support_claim_path(claim), no_visited_state: true %>
+      <%= govuk_link_to t(".claim_title", reference: claim.reference, school_name: claim.school.name), href, no_visited_state: true %>
```
[the remark this answers is not in the document]
Comment (Nitemaeric) [reply app/components/claim/card_component.html.erb:4]: I'll add this in.

**Comment by Nitemaeric** [comment]
[[Payments] Index page](https://trello.com/c/WupnIMqf/256-payments-index-page)

**Pull request merged**